packages/langium/src/lsp/default-lsp-module.ts

```diff
@@ -28,7 +28,7 @@ import { DefaultWorkspaceSymbolProvider } from './workspace-symbol-provider.js';
  * Context required for creating the default language-specific dependency injection module.
  */
 export interface DefaultModuleContext extends DefaultCoreModuleContext {
-    shared: LangiumSharedServices;
+    readonly shared: LangiumSharedServices;
```

Comment:
Since you are throwing an Error on the setter of the proxy, would it be a good idea to require that all properties of the Module are readonly?

Reply:
I'm not sure it's possible to enforce this on the type level in TypeScript.